homeassistant/components/tuya/light.py

```diff
@@ -663,8 +663,11 @@ def turn_on(self, **kwargs: Any) -> None:
                 },
             ]
 
-        elif ATTR_BRIGHTNESS in kwargs and self._brightness:
-            brightness = kwargs[ATTR_BRIGHTNESS]
+        elif (ATTR_BRIGHTNESS in kwargs or ATTR_WHITE in kwargs) and self._brightness:
+            brightness_attr = (
+                ATTR_BRIGHTNESS if ATTR_BRIGHTNESS in kwargs else ATTR_WHITE
+            )
+            brightness = kwargs[brightness_attr]
```

Comment:
Small tweak:
- in the main condition, first check the dpcode, then the attribute
- in the inner condition, drop the single-use variable

```suggestion
        elif self._brightness and (ATTR_BRIGHTNESS in kwargs or ATTR_WHITE in kwargs):
            if ATTR_BRIGHTNESS in kwargs:
                brightness = kwargs[ATTR_BRIGHTNESS]
            else:
                brightness = kwargs[ATTR_WHITE]
```